sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Core
+{
+    /// <summary>
+    ///   A pool of <see cref="TransportProducer" />.
+    ///   It periodically checks if the producers are active and if not it disposes them.
+    /// </summary>
+    ///
+    internal class TransportProducerPool : IAsyncDisposable
+    {
+        /// <summary>The period in minutes after which <see cref="PerformExpiration"/> is run</summary>
+        private const int PerformExpirationPeriodAsMinutes = 10;
```

Comment:
I wonder if we may want to offer this as a `TimeSpan` argument in the constructor that gets defaulted.   Since its an internal class, there's no concern over trying to keep the constructor dead-simple and being able to clearly inject would potentially be helpful for testing.

Reply:
🤔 

```csharp
public TransportProducerPool(ConcurrentDictionary<string, PoolItem> pool = default,
                             TimeSpan? expirationInterval = default)
{
    expirationInterval ??= TimeSpan.FromMinutes(PerformExpirationPeriodAsMinutes);
    Pool = pool ?? new ConcurrentDictionary<string, PoolItem>();

    ExpirationTimer = new Timer(PerformExpiration(),
                                null,
                                expirationInterval.Value,
                                expirationInterval.Value);
}
```

I used ```TimeSpan?``` instead of ```TimeSpan``` as ```default(TimeSpan)``` would be ```0``` and would still a be valid input for the ```Timer```. I would be curious to know what is your thinking on this?

The "price" to pay was adding the ```expirationInterval.Value``` when instantiating the ```Timer```. But we could also use the new null-coalescing assignment operator kindly provided to us with C# 8. It would not be usable with value types such as the ```TimeSpan``` struct.